### PR TITLE
Feature/login screen backend

### LINKED ISSUE #2: Implement Login screen
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Login uses the backend to check the user's credentials (username, password) and forwards the user to another screen. The current user information is stored so that other components can access the user information.

**Describe alternatives you've considered**
None

**Additional context**
Login works with JWT access and refresh tokens. Need to be securely stored (SecureStorage).
Try to use Storybook for this component.

IMG_0346

### PR DESCRIPTION
**Pull request by alexw-at**

### CODE REVIEW

**Review by valentingc — APPROVED** [review]
Great work!

File: src/services/i18n.ts [hunk 1]
```diff
@@ -12,6 +12,15 @@ export const resources = {
...
             username: "Username",
             startJourney: ["Start Journey", "Lets go"],
+            displayName: "Display Name",
+            error: {
+                network: "No Connection 😢",
+                credentials: "Invalid Credentials",
+                username: "Invalid Username",
+                password: "Invalid Password",
+                takenEmailUsername:
+                    "The Email or username is already taken, please choose another one",
```
Comment (DominicLuidold) [inline src/services/i18n.ts]: Inconsistent handling of capitalising nouns ("Email" is capitalised, "username" isn't). Looking at other i18n strings, we might need to think about this.

File: src/services/i18n.ts [hunk 2]
```diff
@@ -35,6 +44,22 @@ export const resources = {
                     "TheLegend27@mail.com",
                 ],
             },
+            validation: {
+                emailRequired: "Please enter a valid email address!",
```
Comment (DominicLuidold) [inline src/services/i18n.ts]: Non-capitalised "email" here for example..

File: App.tsx [hunk 3]
```diff
@@ -1,91 +1,73 @@
...
                 </NavigationContainer>
-            </SafeAreaProvider>
-        </ApolloProvider>
+            )}
+        </SafeAreaProvider>
     );
 }
+function getCurrentUser() {
+    throw new Error("Function not implemented.");
+}
```
Comment (DominicLuidold) [inline App.tsx]: Can we remove this?

File: src/components/ApolloConnection/ApolloConnection.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        return client
+            .mutate({ mutation: REFRESH_TOKEN_MUTATION, errorPolicy: "all" })
+            .then((response) => {
+                // extract your accessToken from your response data and return it
+
+                console.log("fetch token");
+                const data = response.data?.refreshToken;
+                if (!data) {
+                    throw new Error("could not get a refresh token");
```
Comment (DominicLuidold) [inline src/components/ApolloConnection/ApolloConnection.tsx:45]: How is such an error handled within the UI?

**Review by DominicLuidold — CHANGES_REQUESTED** [review]
Please see the notes regarding the inconsistent handling of i18n strings and the one function we might be able to remove..

File: src/components/ApolloConnection/ApolloConnection.tsx (hunk 4 above)
Comment (valentingc) [reply src/components/ApolloConnection/ApolloConnection.tsx:45]: This isn't necessary as the user doesn't need to know what happens technically here.
When the error is catched (line 81), the user is redirected to the Login screen, which should be enough for now.

https://github.com/vacatius/mobile-app/blob/ba8acd98fc4c67e02f12d2782f0cd971f7433d00/src/components/ApolloConnection/ApolloConnection.tsx#L81-L89

Comment (DominicLuidold) [reply src/components/ApolloConnection/ApolloConnection.tsx:45]: I must've missed that. Thanks!

**Comment by valentingc** [comment]
@DominicLuidold resolved the mentioned i18n issues. General i18n concept is IMHO something out of scope of this PR, I'd appreciat this PR getting merged so I can get started with other issues. I cannot proceed as long as this PR is not through due to the refresh token handling, as access tokens are now valid only for a few minutesl

**Review by DominicLuidold — APPROVED** [review]

**Pull request merged**